### PR TITLE
add digitalocean_incidents

### PR DESCRIPTION
**Pull request by circa10a**
Add a gauge to monitor the number of active incidents with digitalocean services.
Sourced from [their status api](https://status.digitalocean.com/api/v2)

### CODE REVIEW

**Comment by metalmatze** [comment]
Nice! That's definitely super handy!
Is that API not part of godo?

**Comment by circa10a** [comment]
@metalmatze Sadly, no. That would have made things much simpler!

Their status page service seemed to be the only place, but it has a really great API.

**Comment by metalmatze** [comment]
Ok. Not a blocker at all. Just wanted to check. I'll give it a proper review soon.

File: collector/incidents.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	logger  log.Logger
+	errors  *prometheus.CounterVec
+	client  *godo.Client
+	timeout time.Duration
+
+	Incidents *prometheus.Desc
+}
+
+// NewIncidentCollector returns a new IncidentCollector.
+func NewIncidentCollector(logger log.Logger, errors *prometheus.CounterVec, client *godo.Client, timeout time.Duration) *IncidentCollector {
```
Comment (metalmatze) [inline collector/incidents.go:29]: No need to pass the `cilent *godo.Client` if it's not used further down.

File: collector/incidents.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	defer r.Body.Close()
+	if r.StatusCode == http.StatusOK {
+		bodyBytes, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return []gjson.Result{}, err
+		}
+		return gjson.Get(string(bodyBytes), "incidents.#.name").Array(), nil
+	}
+	return []gjson.Result{}, errors.New("Unable to retrieve incidents")
+}
```
Comment (metalmatze) [inline collector/incidents.go:68]: What do you think about these changes?
I don't think we necessarily need the gjson dependency. :blush: 
```go
func GetIncidents(client *http.Client) (int, error) {
	r, err := client.Get(doStatusAPIURL)
	if err != nil {
		return 0, err
	}
	defer r.Body.Close()

	if r.StatusCode != http.StatusOK {
		return 0, fmt.Errorf("unable to retrieve incidents: %w", err)
	}

	type payload struct {
		Incidents []interface{} `json:"incidents"`
	}

	var p payload
	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
		return 0, err
	}

	return len(p.Incidents), nil
}
```

File: collector/incidents.go (hunk 1 above)
Comment (circa10a) [reply collector/incidents.go]: 👍

**Comment by circa10a** [comment]
@metalmatze Addressed both of the issues and added the ability to filter by region. You can test the regional functionality by changing the API URL to the incident history endpoint:

`const doStatusAPIURL = "https://s2k7tnzlhrpw.statuspage.io/api/v2/incidents.json"`

**Comment by circa10a** [comment]
@metalmatze just a gentle reminder 😇

**Comment by metalmatze** [comment]
@circa10a Thanks.
I just started some free time from work, so maybe in the next week I have some time to work on it. 😊

**Pull request merged**